src/main/java/domain/board/Board.java

```diff
@@ -0,0 +1,145 @@
+package domain.board;
+
+import domain.piece.Color;
+import domain.piece.Pawn;
+import domain.piece.Piece;
+import java.util.Map;
+
+public class Board {
+
+    private final Map<Position, Piece> board;
+
+    private Board(Map<Position, Piece> board) {
+        this.board = board;
+    }
+
+    public static Board generatedBy(BoardGenerator boardGenerator) {
+        return new Board(boardGenerator.generate());
+    }
+
+    public void move(Position sourcePosition, Position targetPosition, Color color) {
+        Piece piece = board.get(sourcePosition);
+        validate(sourcePosition, targetPosition, color);
+        board.remove(sourcePosition);
+        board.put(targetPosition, piece);
+    }
+
+    private void validate(Position sourcePosition, Position targetPosition, Color color) {
+        validateNoPieceToMove(sourcePosition);
+        validateTurn(sourcePosition, color);
+        validateSamePosition(sourcePosition, targetPosition);
+        validateOwnPieceExistAtTargetPosition(sourcePosition, targetPosition);
+        validatePieceCanMove(sourcePosition, targetPosition);
+        validateWhenStraightOrDiagonalMove(sourcePosition, targetPosition);
+        validateWhenPieceIsPawn(sourcePosition, targetPosition);
+    }
+
+    private void validateNoPieceToMove(Position sourcePosition) {
+        if (isNoPieceAt(sourcePosition)) {
+            throw new IllegalArgumentException("source 위치에 말이 없습니다.");
+        }
+    }
+
+    private void validateTurn(Position sourcePosition, Color color) {
+        Piece piece = board.get(sourcePosition);
+        if (piece.hasColorOf(color)) {
+            return;
+        }
+        throw new IllegalArgumentException("자신의 말만 움직일 수 있습니다.");
+    }
```

Comment:
```suggestion
    private void validateTurn(Position sourcePosition, Color color) {
        Piece piece = board.get(sourcePosition);
        if (!piece.hasColorOf(color)) {
            throw new IllegalArgumentException("자신의 말만 움직일 수 있습니다.");
        }
    }
```
throw 를 먼저하고, return 을 마지막에 하는 구조를 유지해주세요. 프로젝트 내의 다른 validate 방식도 확인해주시기 바랍니다.

Reply:
throw를 먼저하는 것이 검증을 먼저하고 불필요한 코드를 줄일 수 있는 것 같습니다!

```
private void validatePieceCanMove(Position source, Position target) {
        Piece piece = board.get(source);
        if (!piece.canMove(source, target)) {
            throw new IllegalArgumentException("말의 규칙에 맞지 않는 이동입니다.");
        }
    }
```
해당 코드 같은 경우 canMove가 추상 메서드로 정의되어 있어 cannotMove를 정의하는데 필요한 리소스가 크다고 생각했습니다. 따라서 canMove에 ! 를 붙여 조건문을 작성했습니다.
조건에 !를 붙이는 대신 새로운 메서드를 정의하여 가독성을 높이라는 피드백을 받았었는데, 지금 같은 경우에는 어쩔 수 없이 포기해야하는 부분인지, 아니면 제가 놓친 부분이 있는지 궁금합니다!

Reply:
! 연산자를 사용할지, 메서드를 따로 만들지는 개발자마다 모두 의견이 달라요. 그래서 이건 정말 선호도의 차이, 팀의 컨벤션 차이일 것 같네요. 그래서 이와 관련 리뷰는 남기지 않아요.

저는 굳이 별도의 메서드를 만들지 않고 ! 연산자를 사용해도 된다고 생각합니다. 물론 if 문 안에 들어가는 조건이 매우 복잡하고 길어진다면 ! 표시를 못 볼 수 있겠지만, 딱히 그런 상황이 아니라서 가독성이 크게 하락하는 것 같진 않아요. 만약 반대 상황이 자주 사용되거나 조건이 매우 길어져서 연산자를 보기 어렵다면 그 때 추가하는걸 고려해봐도 괜찮겠네요.

가장 중요한건 무엇보다 '컨벤션' 이라고 생각해요. 만약 메서드를 따로 만들기로 했다면 모든 팀원이 그 방식을 따라야한다고 생각합니다. 어떨 때는 메서드를 사용하고 또 다른 상황에서는 ! 연산자를 사용한다면 통일성이 없어지고, 어디서 그 로직이 사용되는지 제대로 파악하기 힘드니까요.

따라서 카피의 경우에는 그냥 ! 연산자 사용해도 큰 문제 없을 것 같아요!

Reply:
! 연산자를 사용할지, 메서드를 따로 만들지는 개발자마다 모두 의견이 달라요. 그래서 이건 정말 선호도의 차이, 팀의 컨벤션 차이일 것 같네요.

저는 굳이 별도의 메서드를 만들지 않고 ! 연산자를 사용해도 된다고 생각합니다. 물론 if 문 안에 들어가는 조건이 매우 복잡하고 길어진다면 ! 표시를 못 볼 수 있겠지만, 딱히 그런 상황이 아니라서 가독성이 크게 하락하는 것 같진 않아요. 만약 반대 상황이 자주 사용되거나 조건이 매우 길어져서 연산자를 보기 어렵다면 그 때 추가하는걸 고려해봐도 괜찮겠네요.

가장 중요한건 무엇보다 '컨벤션' 이라고 생각해요. 만약 메서드를 따로 만들기로 했다면 모든 팀원이 그 방식을 따라야한다고 생각합니다. 어떨 때는 메서드를 사용하고 또 다른 상황에서는 ! 연산자를 사용한다면 통일성이 없어지고, 어디서 그 로직이 사용되는지 제대로 파악하기 힘드니까요.

따라서 카피의 경우에는 그냥 ! 연산자 사용해도 큰 문제 없을 것 같아요!